lib/Processor.php

```diff
@@ -890,78 +866,28 @@ private function compilePacket() {
             fwrite(STDERR, ob_get_clean());
         }
 
+        $this->seqId = -1;
+
         return $packet;
     }
 
-    private function compressPacket($packet) {
-        $packet = $this->uncompressedOut.$packet;
-
+    private function compressPacket(string $packet): string {
         if ($packet == "") {
             return "";
         }
 
         $len = strlen($packet);
-        while ($len < self::MAX_UNCOMPRESSED_BUFLEN && !empty($this->out)) {
```

Comment:
self::MAX_UNCOMPRESSED_BUFLEN is important to ensure that packets larger than this are properly split up - you've just straight ahead removed the whole logic for that.